bfit/density.py

```diff
@@ -434,14 +543,23 @@ def positive_definite_kinetic_energy(self, points):
             phi_matrix[:, index] = np.ravel(np.dot(deriv_radial, self.orbitals_coeff[orbital])**2.0)
 
             # Calculate del^2 of spherical component
-            slater = SlaterAtoms.radial_slater_orbital(exps, numbers, points)
-            with np.errstate(divide='ignore'):
-                gradient_sph = (
-                    angular[orbital[1]] *
-                    np.ravel(np.dot(slater, self.orbitals_coeff[orbital]))**2.0 /
-                    points**2.0
-                )
-                gradient_sph[np.abs(points) < 1e-10] = 0.0
+            norm = np.power(2. * exps, numbers) * np.sqrt((2. * exps) / factorial(2. * numbers))
+            # Take unnormalized slater with number n-1, this is needed to remove divide by r^2
+            slater_minus_one = SlaterAtoms.radial_slater_orbital(
+                exps, numbers - 1, points, normalized=False
+            )
+            deriv_pref = norm.T * slater_minus_one
+            # When r=0 and n = 1, then the derivative is undefined and this returns infinity or nan
+            i_r_zero = np.where(np.abs(points) == 0.0)[0]
+            i_numb_one = np.where(numbers[0] == 1)[0]
+            indices = np.array([[x, y] for x in i_r_zero for y in i_numb_one])
+            if len(indices) != 0:  # if-statement needed to remove numpy warning using list
+                deriv_pref[indices] = np.inf
```

Comment:
@Ali-Tehrani can this be computed using `first_derivative_radial_slater_type_orbital`?